components/list/src/options.rs

```diff
@@ -0,0 +1,55 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+/// A list of options set by the developer to adjust the behavior of the ListFormatter.
+///
+/// # Examples
+/// ```
+/// use icu::list::{ListFormatterOptions, ListLength};
+///
+/// let options = ListFormatterOptions::default()
+///     .with_length(ListLength::Wide);
+/// ```
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[non_exhaustive]
+pub struct ListFormatterOptions {
+    /// The length variant should reflect available space for the list.
+    pub length: Option<ListLength>,
+}
+
+impl Default for ListFormatterOptions {
+    fn default() -> Self {
+        Self::default()
+    }
+}
+
+impl ListFormatterOptions {
+    /// Constructs a new [`ListFormatterOptions`] struct.
+    pub const fn default() -> Self {
+        Self { length: None }
+    }
+
+    /// Auguments the struct with the set [`ListLength`].
+    pub const fn with_length(mut self, length: ListLength) -> Self {
+        self.length = Some(length);
+        self
+    }
```

Comment:
Suggestion for a follow-up: we could add functions here `pub const fn long() -> Self` so that you don't need to import both `ListLength` and `ListFormatterOptions`. I did this in icu_datetime.

Reply:
My concern is that it doesn't scale. I aspire to have our constructors be predictable over convenient. 
I realize that it's a preference and we may not be aligned on that.

Reply:
I consider length enums to be special because they are so common. I wouldn't suggest this for other options, necessarily.